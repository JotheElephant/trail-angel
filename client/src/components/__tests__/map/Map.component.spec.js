'use strict';

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Map from '../../map/Map.component';
import homeData from '../../../../__tests__/fixtures/home-data';

jest.mock('react-native-maps');
jest.mock('../../../api/google-api');
jest.mock('../../../api/trailangel-api', () => {
  return {
    getGeo: () => {
      return Promise.resolve();
    }
  }
});
jest.mock('../../../utils/trail-calculations');

describe('Map component', () => {
  it('renders correctly', () => {
    const data = homeData[0];
    const tree = renderer.create(
      <Map {...data} />
    );
    expect(tree).toMatchSnapshot();
  });
});