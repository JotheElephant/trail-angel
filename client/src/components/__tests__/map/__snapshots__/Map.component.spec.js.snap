exports[`Map component renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "height": 32,
          "marginTop": 42,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.7)",
            "borderRadius": 5,
            "marginRight": 5,
            "opacity": 1,
            "paddingBottom": 3,
            "paddingHorizontal": 8,
            "paddingTop": 6,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          0.0 mi
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.7)",
            "borderRadius": 5,
            "marginRight": 5,
            "opacity": 1,
            "paddingBottom": 3,
            "paddingHorizontal": 8,
            "paddingTop": 6,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          NaN ft
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.7)",
            "borderRadius": 5,
            "marginRight": 5,
            "paddingBottom": 3,
            "paddingHorizontal": 8,
            "paddingTop": 6,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail" />
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": -64,
        }
      }>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.7)",
              "borderRadius": 5,
              "marginRight": 5,
              "opacity": 1,
              "paddingBottom": 3,
              "paddingHorizontal": 8,
              "paddingTop": 6,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Remove Last Pin
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.7)",
              "borderRadius": 5,
              "marginRight": 5,
              "opacity": 1,
              "paddingBottom": 3,
              "paddingHorizontal": 8,
              "paddingTop": 6,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Delete
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.7)",
              "borderRadius": 5,
              "marginRight": 5,
              "opacity": 1,
              "paddingBottom": 3,
              "paddingHorizontal": 8,
              "paddingTop": 6,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                },
                Object {
                  "opacity": 0.5,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.7)",
              "borderRadius": 5,
              "marginRight": 5,
              "opacity": 1,
              "paddingBottom": 3,
              "paddingHorizontal": 8,
              "paddingTop": 6,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                },
                Object {
                  "opacity": 1,
                },
                Object {
                  "fontFamily": "fontcustom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
